### PR TITLE
Make logs friendlier when agent fails to connect to the manager

### PR DESCRIPTION
**Pull request by aritosteles**
|Related issue|
|---|
|[255](https://github.com/wazuh/wazuh-agent/issues/255)|

## Description
Make output more friendly when the agent fails to connect to the manager

## Tests

- Compilation without warnings in every supported platform
  - [x] Linux
  - [ ] Windows
  - [ ] MAC OS X

### CODE REVIEW

File: src/agent/src/unix_daemon.cpp [hunk 1]
```diff
@@ -86,7 +86,6 @@ namespace unix_daemon
 
         if (!file.is_open())
         {
-            LogCritical("Error opening file: {}({}) {}", filename.c_str(), errno, std::strerror(errno));
```
Comment (TomasTurina) [inline src/agent/src/unix_daemon.cpp:89]: What about [line 97](https://github.com/wazuh/wazuh-agent/pull/273/files#diff-feb07c04fa143494f660f59effee9bd4054b45fc697cec7061e609fd26576c70R97)? Is it a critical log?

File: src/agent/src/unix_daemon.hpp [hunk 2]
```diff
@@ -18,16 +20,34 @@ namespace unix_daemon
         PIDFileHandler(PIDFileHandler&&) = default;
         PIDFileHandler& operator=(PIDFileHandler&&) = default;
 
+        /// @brief Reads the PID file and returns its value.
+        /// @return The value read from the PID file or 0 if the file does not exist.
         static pid_t ReadPIDFromFile();
 
     private:
+        /// @brief  Creates a directory relative to the current path.
```
Comment (TomasTurina) [inline src/agent/src/unix_daemon.hpp:28]:
```suggestion
        /// @brief Creates a directory relative to the current path.
```

File: src/agent/src/unix_daemon.cpp (hunk 1 above)
Comment (aritosteles) [reply src/agent/src/unix_daemon.cpp:89]: Yes, the file has already been opened but it is failing to read

Comment (TomasTurina) [reply src/agent/src/unix_daemon.cpp:89]: But a critical error means that the execution exits. I'm not sure if that error means that.

File: src/agent/communicator/src/communicator.cpp [hunk 3]
```diff
@@ -119,12 +121,12 @@ namespace communicator
...
                 if (result != boost::beast::http::status::ok)
                 {
-                    return std::chrono::milliseconds(A_SECOND_IN_MILLIS);
+                    return std::chrono::milliseconds(CONNECTION_RETRY_MSECS);
                 }
                 else
                 {
                     return std::chrono::milliseconds((GetTokenRemainingSecs() - TOKEN_PRE_EXPIRY_SECS) *
-                                                     A_SECOND_IN_MILLIS);
+                                                     1000); // NOLINT
```
Comment (jr0me) [inline src/agent/communicator/src/communicator.cpp:129]: Instead of hardcoding the `1000` and adding the `NOLINT` keep the variable this PR is removing (`A_SECOND_IN_MILLIS`). There are a few occurrences of this in this file and others.

File: src/agent/src/unix_daemon.cpp (hunk 1 above)
Comment (aritosteles) [reply src/agent/src/unix_daemon.cpp:89]: We could just return 0.

This function is called
- during startup to check for a running instance
- when the agent is run with `--stop`
- when the agent is run with `--status`

We need to determine what having a PID file and not being able to read it means (if at all possible), and what to do in each scenario.

File: src/agent/communicator/src/communicator.cpp (hunk 3 above)
Comment (aritosteles) [reply src/agent/communicator/src/communicator.cpp]: Done

File: src/agent/communicator/src/communicator.cpp [hunk 4]
```diff
@@ -47,6 +46,9 @@ namespace communicator
             {
                 LogInfo("Using insecure connection.");
             }
+
+            std::string retryIntervalSecs = getStringConfigValue("agent", "retry_interval_secs");
```
Comment (TomasTurina) [inline src/agent/communicator/src/communicator.cpp:50]: Maybe you can simplify these two lines into one.

File: src/agent/communicator/src/communicator.cpp [hunk 5]
```diff
@@ -119,12 +123,12 @@ namespace communicator
                 const auto result = SendAuthenticationRequest();
                 if (result != boost::beast::http::status::ok)
                 {
-                    return std::chrono::milliseconds(A_SECOND_IN_MILLIS);
+                    return std::chrono::milliseconds(m_retryIntervalSecs * A_SEC_IN_MSECS);
```
Comment (TomasTurina) [inline src/agent/communicator/src/communicator.cpp:126]: Why did you change this?

File: src/agent/communicator/src/http_client.cpp [hunk 6]
```diff
@@ -152,7 +157,7 @@ namespace http_client
             LogDebug("Response code: {}.", res.result_int());
             LogDebug("Response body: {}.", boost::beast::buffers_to_string(res.body().data()));
 
-            const auto duration = std::chrono::milliseconds(1000);
+            const auto duration = std::chrono::milliseconds(CONNECTION_RETRY_MSECS);
```
Comment (TomasTurina) [inline src/agent/communicator/src/http_client.cpp:160]: Why did you change this?

File: src/agent/communicator/src/http_client.cpp [hunk 7]
```diff
@@ -10,6 +10,8 @@
 
 namespace http_client
 {
+    constexpr int CONNECTION_RETRY_MSECS = 5000;
```
Comment (TomasTurina) [inline src/agent/communicator/src/http_client.cpp:13]: This shouldn't exist, now we have the configurable setting.

File: src/agent/src/process_options_unix.cpp [hunk 8]
```diff
@@ -23,11 +23,16 @@ void RestartAgent(const std::string& configFile)
...
     pid_t pid = unix_daemon::PIDFileHandler::ReadPIDFromFile();
-    if (pid != 0)
+    if (pid > 0)
     {
         std::cout << "wazuh-agent already running\n";
         return;
     }
+    else if (pid < 0)
+    {
+        std::cout << "Error reading pid file\n";
```
Comment (TomasTurina) [inline src/agent/src/process_options_unix.cpp:33]:
```suggestion
        std::cout << "Error reading pid file\n";
        return;
```

File: src/agent/communicator/src/communicator.cpp (hunk 4 above)
Comment (aritosteles) [reply src/agent/communicator/src/communicator.cpp:50]: Done

File: src/agent/communicator/src/communicator.cpp [hunk 9]
```diff
@@ -27,7 +27,7 @@
 namespace communicator
 {
     constexpr int TOKEN_PRE_EXPIRY_SECS = 2;
-    constexpr int A_SECOND_IN_MILLIS = 1000;
+    constexpr int A_SEC_IN_MSECS = 1000;
```
[the remark this answers is not in the document]
Comment (aritosteles) [reply src/agent/communicator/src/communicator.cpp:30]: Done

File: src/agent/communicator/src/communicator.cpp (hunk 5 above)
Comment (aritosteles) [reply src/agent/communicator/src/communicator.cpp:126]: This was mistaken. Fixed

File: src/agent/src/process_options_unix.cpp (hunk 8 above)
Comment (aritosteles) [reply src/agent/src/process_options_unix.cpp:33]: Done

File: src/agent/communicator/src/http_client.cpp (hunk 7 above)
Comment (aritosteles) [reply src/agent/communicator/src/http_client.cpp]: That's what I was asking earlier. Wasn't sure whether to use the same setting. Done.

**Review by vikman90 — APPROVED** [review]

**Review by sdvendramini — APPROVED** [review]

**Review by jr0me — APPROVED** [review]

**Review by TomasTurina — APPROVED** [review]
